decoder/DecoderApi_unittest.cpp

```diff
@@ -273,15 +273,16 @@ TEST_P(DecodeApiTest, Format_Change)
     SharedPtr<IVideoDecoder> decoder;
     TestDecodeFrames frames = *GetParam();
     decoder.reset(createVideoDecoder(frames.getMime()), releaseVideoDecoder);
-    ASSERT_TRUE(bool(decoder));
 
     decoder->setAllocator(allocator);
+    ASSERT_TRUE(bool(decoder));
```

Comment:
Still not quite right... this will crash if decoder is invalid (at `decoder->setAllocator(...)`).  The `allocator` should be created and set **_after_** asserting the decoder is valid, like this:

```
decoder.reset(createVideoDecoder(frames.getMime()), releaseVideoDecoder);
ASSERT_TRUE(bool(decoder));
DecodeSurfaceAllocator* allocator = new DecodeSurfaceAllocator(display);
decoder->setAllocator(allocator);
```

Reply:
yeah, you are right, it should be confirmed whether decoder is invalid or not, then we can use it.